File: leetcode/medium/38. Count and Say/solution.js
/** https://leetcode.com/problems/count-and-say/
 * @param {number} n
 * @return {string}
 */
function countAndSay(n) {
    if (n === 1) return '1';

    let str = '1';
    for (let i = 1; i < n; i += 1) {
        let count = 1, buffer = '';

        for (let j = 1; j <= str.length; j += 1) {
            const prev = str[j-1], curr = str[j];
            if (prev === curr) {
                count += 1;
            }
            else {
                buffer += count + prev;
                count = 1;
            }
        }
        str = buffer;
    }
    return str;
};

File: leetcode/medium/38. Count and Say/solution.cc
#include <string>
#include <iostream>

using namespace std;

class Solution {
public:
    string countAndSay(int n) {
        if (n == 1) return "1";

        string str = countAndSay(n - 1);

        int count = 0;
        char prev;
        string buffer = "";

        for (char ch: str) {
            if (ch == prev) {
                count++;
            }
            else {
                if (count != 0) {
                    buffer += to_string(count) + prev;
                }
                prev = ch;
                count = 1;
            }
        }

        if (count != 0) {
            buffer += to_string(count) + prev;
        }
    return buffer;
    }
};

int main() {
    Solution sol;
    cout << sol.countAndSay(6);
}